From c542703f06759de8982e37d8b3799a81a2fec60b Mon Sep 17 00:00:00 2001
From: Huang Yang <huangyang@loongson.cn>
Date: Wed, 25 May 2022 01:47:26 +0000
Subject: [PATCH] 389-ds-base 1.4.3.20 add loongarch support

---
 include/jemalloc/internal/quantum.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/jemalloc/internal/quantum.h b/include/jemalloc/internal/quantum.h
index 821086e..9428a34 100644
--- a/include/jemalloc/internal/quantum.h
+++ b/include/jemalloc/internal/quantum.h
@@ -61,6 +61,9 @@
 #  ifdef __le32__
 #    define LG_QUANTUM		4
 #  endif
+#  ifdef __loongarch__
+#    define LG_QUANTUM         4
+#  endif
 #  ifndef LG_QUANTUM
 #    error "Unknown minimum alignment for architecture; specify via "
 	 "--with-lg-quantum"
-- 
2.33.0

